packages/next/src/server/dev/turbopack-utils.ts

```diff
@@ -233,48 +233,110 @@ export async function readPartialManifest<T>(
     | `${typeof NEXT_FONT_MANIFEST}.json`
     | typeof REACT_LOADABLE_MANIFEST,
   pageName: string,
-  type:
-    | 'pages'
-    | 'app'
-    | 'app-route'
-    | 'middleware'
-    | 'instrumentation' = 'pages'
+  type: 'pages' | 'app' | 'middleware' | 'instrumentation' = 'pages'
 ): Promise<T> {
   const manifestPath = posix.join(
     distDir,
     `server`,
-    type === 'app-route' ? 'app' : type,
+    type,
     type === 'middleware' || type === 'instrumentation'
       ? ''
-      : pageName === '/'
-      ? 'index'
-      : pageName === '/index' || pageName.startsWith('/index/')
-      ? `/index${pageName}`
-      : pageName,
-    type === 'app' ? 'page' : type === 'app-route' ? 'route' : '',
+      : type === 'app'
+      ? pageName
+      : getAssetPathFromRoute(pageName),
     name
   )
   return JSON.parse(await readFile(posix.join(manifestPath), 'utf-8')) as T
 }
 
-export type BuildManifests = Map<string, BuildManifest>
-export type AppBuildManifests = Map<string, AppBuildManifest>
-export type PagesManifests = Map<string, PagesManifest>
-export type AppPathsManifests = Map<string, PagesManifest>
-export type MiddlewareManifests = Map<string, TurbopackMiddlewareManifest>
-export type ActionManifests = Map<string, ActionManifest>
-export type FontManifests = Map<string, NextFontManifest>
-export type LoadableManifests = Map<string, LoadableManifest>
-export type CurrentEntrypoints = Map<string, Route>
+/**
+ * `app` -> app dir
+ * `pages` -> pages dir
+ * `root` -> middleware / instrumentation
+ * `assets` -> assets
+ */
+export type EntryKeyType = 'app' | 'pages' | 'root' | 'assets'
+export type EntryKeySide = 'client' | 'server'
+
+export type EntryKey = `${EntryKeyType}@${EntryKeySide}@${string}`
+
+/**
+ * Get a key that's unique across all entrypoints.
+ */
+export function getEntryKey<
+  Type extends EntryKeyType,
+  Side extends EntryKeySide,
+  Page extends string
+>(type: Type, side: Side, page: Page): `${Type}@${Side}@${Page}` {
+  return `${type}@${side}@${page}` satisfies EntryKey
```

Comment:
How about using JSON?

```suggestion
  return JSON.stringify({ type, side, page })
```

Reply:
done